src/app/api/chat/[id]/route.ts

```diff
@@ -20,6 +22,8 @@ export async function GET(
   }
 
   const otherUserId = params.id;
+  initializeWSServer();
```

Comment:
Initialising WS server in API route is not ideal. Few reasons:
1. API routes are meant to be stateless, should not manage long-lived WS connections.
2. Initialising WS server every GET request is inefficient.
3. This may create multiple WS servers; WS server should be a singleton.